blog/2020-10-09-hs-to-coq/2020-10-09-hs-to-coq.md

```diff
@@ -0,0 +1,1172 @@
+---
+title: "Tutorial: Verify Haskell Programs with hs-to-coq"
+author: Li-yao Xia
+description: A tutorial on hs-to-coq
+tags: haskell, coq
+withtoc: true
+alectryon: [-Q, "blog/2020-10-09-hs-to-coq/lib/Src", Src, -Q, "blog/2020-10-09-hs-to-coq/lib/Proofs", Proofs]
+alectryon-cache: "blog/2020-10-09-hs-to-coq/alectryon/"
+---
+
+## Preface
+
+Programming is the power of automation at our fingertips.
+A task, a program, can be run over and over, billions of instructions
+per second. And with great power comes great responsibility.
+A program can also fail, over and over, billions of instructions per second.
+Many solutions have been devised to reduce the likelihood and impact of human
+error in programming, testing and type systems being prominent examples.
+Programming is also a domain where pure logic prospers especially well.
+Programs are mathematical objects that we can prove properties about,
```

Comment:
This sentence has too many subjects, I think.

Reply:
Maybe I should just cut the last two sentences to say something like "Another approach is formal verification."?

Reply:
I think that would be more direct and better!